Sources/DriverSupport/DriverSupportUtils.swift

```diff
@@ -0,0 +1,59 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftDriver
+import class TSCBasic.Process
+import enum TSCBasic.ProcessEnv
+import struct TSCBasic.ProcessResult
+
+import protocol TSCBasic.FileSystem
+
+public class DriverSupport {
+    var supportedDriverFlags: Set<String>?
+    public init() {}
+
+    // This checks supported _frontend_ flags, which are not necessarily supported in the driver.
+    public static func checkSupportedFrontendFlags(flags: Set<String>, fileSystem: FileSystem) -> Bool {
+        do {
+            let executor = try SPMSwiftDriverExecutor(resolver: ArgsResolver(fileSystem: fileSystem), fileSystem: fileSystem, env: [:])
+            let driver = try Driver(args: ["swiftc"], executor: executor)
+            return driver.supportedFrontendFlags.intersection(flags) == flags
+        } catch {
+            return false
+        }
+    }
+
+    // Currently there's no good way to get supported flags from the built-in toolchain driver, so call `swiftc -h` directly
+    // and save the result so we don't spawn processes repeatedly.
+    public func checkSupportedDriverFlags(flags: Set<String>) -> Bool {
```

Comment:
should we add a cache here so we dont have to shell out frequently to get the same information?